aerich/migrate.py

```diff
@@ -190,18 +192,22 @@ def _add_operator(cls, operator: str, upgrade=True, fk_m2m_index=False) -> None:
     @classmethod
     def _handle_indexes(cls, model: Type[Model], indexes: List[Union[Tuple[str], Index]]) -> list:
         ret: list = []
+        for index in indexes:
+            if (
+                isinstance(index, Index)
+                and (index_cls := type(index)) not in cls._patched_index_classes
+            ):
 
-        def index_hash(self) -> str:
-            h = hashlib.new("MD5", usedforsecurity=False)  # type:ignore[call-arg]
-            h.update(
-                self.index_name(cls.ddl.schema_generator, model).encode()
-                + self.__class__.__name__.encode()
-            )
-            return h.hexdigest()
+                def _hash(self) -> int:
```

Comment:
Hey, sorry, I missed your question in https://github.com/tortoise/aerich/pull/311#discussion_r1870578779.

It would be definitely better to add __eq__ and __hash__ to indexes in tortoise instead of patching it. Although, these aren't great implementations. I would suggest something like

```python
    def __hash__(self):
        return hash((tuple(self.fields), self.name, tuple(self.expressions)))

    def __eq__(self, other):
        if not isinstance(other, Index):
            return False
        return self.__dict__ == other.__dict__
```

Reply:
1. Agree that add eq/hash in tortoise is better.  So I add a `if Index(fields=('id',) != Index(fields=('id',)` compare to check that whether it is already handled in tortoise. And patch it can fix the issue with previous versions of tortoise.
2. I don't get the point why use `self.__dict__ == other.__dict__` instead of `hash(self) == hash(other)`?